### PR TITLE
Add pkg-config files

### PR DESCRIPTION
**Pull request by DanySpin97**
This PR is based on Fedora patch: https://src.fedoraproject.org/rpms/glslang/blob/master/f/0001-pkg-config-compatibility.patch

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -168,18 +173,10 @@ if(BUILD_EXTERNAL AND IS_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/External)
 	add_subdirectory(External)
 endif()
 
-if(NOT TARGET SPIRV-Tools-opt)
-    set(ENABLE_OPT OFF)
-endif()
-
-if(ENABLE_OPT)
+# This is always true
+if(${SPIRV_TOOLS_FOUND} EQUAL 1)
```
Comment (johnkslang) [inline CMakeLists.txt]: This, or something like it, seems to have broken CMake, at least for AppVeyor.

Comment (DanySpin97) [reply CMakeLists.txt]: AppVeyor build uses Windows and does not ship pkg-config tool. I'm looking for a way to fix this. If Windows cannot be fixed, I'm going to add a check and use the provided SPIRV-Tools on it.

**Comment by DanySpin97** [comment]
Moreover, when -DBUILD_SHARED_LIBS=OFF (which is the defaults), [this fix](https://github.com/KhronosGroup/SPIRV-Tools/pull/3189) for SPIRV-Tools is required.

File: CMakeLists.txt (hunk 1 above)
Comment (DanySpin97) [reply CMakeLists.txt]: I'll open a separate PR to build using system SPIRV-Tools.

File: SPIRV/CMakeLists.txt [hunk 2]
```diff
@@ -40,6 +40,9 @@ target_include_directories(SPIRV PUBLIC
 	$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/..>
 	$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
 
+set(SPIRV_NAME spirv)
+set(SPIRV_VERSION 1.5)
```
Comment (johnkslang) [inline SPIRV/CMakeLists.txt]: Why is this needed?  We seem to be increasing the number of random places that will need to be changed when SPIR-V is updated.

File: glslang/CMakeLists.txt [hunk 3]
```diff
@@ -80,6 +80,7 @@ set(HEADERS
     MachineIndependent/preprocessor/PpContext.h
     MachineIndependent/preprocessor/PpTokens.h)
 
+set(VERSION 8.13.3559)
```
Comment (johnkslang) [inline glslang/CMakeLists.txt]: This number is already out of date...

File: SPIRV/CMakeLists.txt (hunk 2 above)
Comment (DanySpin97) [reply SPIRV/CMakeLists.txt:44]: It is needed to set a version in the `spirv.pc` file. What could be done is to pass SPIRV version from cmake to `spirv.hpp`.

File: glslang/CMakeLists.txt (hunk 3 above)
Comment (DanySpin97) [reply glslang/CMakeLists.txt:83]: Same thing here.

**Comment by johnkslang** [comment]
There is enough feedback here that a better alternate way is needed, so closing this one.  Thanks.
